src/MongoDB.Driver.Core/Core/Compression/Zstd/NativeBufferInfo.cs

```diff
@@ -0,0 +1,29 @@
+﻿/* Copyright 2020–present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+
+using System;
+using System.Runtime.InteropServices;
+
+namespace MongoDB.Driver.Core.Compression.Zstd
+{
+    [StructLayout(LayoutKind.Sequential)]
+    internal class NativeBufferInfo
+    {
+        public IntPtr DataPointer = IntPtr.Zero;
+        public ulong Size = 0;
```

Comment:
As it was discussed in the slack, I've replaced `UIntPtr` on ulong for these properties.